Tests/test_tiff_ifdrational.py

```diff
@@ -43,6 +50,12 @@ def test_nonetype():
     assert xres and yres
 
 
+def test_nan():
+    # usually NaN != NaN, but this would break exif self-equality
+
+    assert IFDRational(123, 0) == IFDRational(123, 0)
```

Comment:
You're not actually loading EXIF values of NaN, right? This is just for theoretical purposes?

Reply:
Actually, it's a very practical issue: I am loading Exif from a [test image](https://raw.githubusercontent.com/luphord/imgwrench/master/tests/images/town.jpg). `exif[34853][6]` (GPS tag altitude) happens to be `0/0`, i.e. NaN. I do not care about this particular value, but I do care about `exif == exif_loaded_again`.

I use this equality assertion in test cases to ensure that loaded Exif data was saved to a transformed image unchanged. Alternatively, I could compare the Exif raw byte sequences. But then again: If the byte sequences are equal, why should the deserialized objects not compare equal?

Reply:
I'm not sold on the idea that NaN should equal NaN. If the GPS altitude is NaN, then one GPS altitude is not equal to the other, so maybe the EXIF data shouldn't be considered equal? I hear what you're saying, I'm just uncomfortable with the idea of causing our subclass of `Rational` to violate a standard like that.

If anyone else has input, feel free to jump in.

Reply:
I see your point and understand that you do not want to deviate from standard behaviour. And maybe this is and edge case with little relevance. Does anyone have experience with NaNs in EXIF data?

Apart from this I wonder if the standard NaN != NaN behaviours is really what you want here:
- Up to version 7.2.0 loading EXIF from the *same image* resulted in equal objects, now it does not anymore.
- Equal EXIF byte streams do not necessarily imply equal objects anymore.

Regardless, I am open to removing the NaN-check from this PR. I believe that fixing `numerator` / `denominator` equality adds value on its own. We would then have equality if there are no NaNs present (which is a reasonably simple statement in my opinion).

Reply:
I've created #4888 to try and get the changes in that are agreed on, at least.